src/Advection/vector_invariant_advection.jl

```diff
@@ -38,142 +32,173 @@ Keyword arguments
 - `vorticity_scheme`: Scheme used for `Center` reconstruction of vorticity, options are upwind advection schemes
                       - `UpwindBiased` and `WENO` - in addition to an `EnergyConservingScheme` and an `EnstrophyConservingScheme`
                       (defaults to `EnstrophyConservingScheme`)
-- `divergence_scheme`: Scheme used for `Face` reconstruction of divergence. Options are upwind advection schemes
-                       - `UpwindBiased` and `WENO` - or `nothing`. In case `nothing` is specified, divergence flux is
-                       absorbed into the vertical advection term (defaults to `nothing`). If `vertical_scheme` isa `EnergyConservingScheme`,
-                       divergence flux is absorbed in vertical advection and this keyword argument has no effect
 - `vorticity_stencil`: Stencil used for smoothness indicators in case of a `WENO` upwind reconstruction. Choices are between `VelocityStencil`
                        which uses the horizontal velocity field to diagnose smoothness and `DefaultStencil` which uses the variable
                        being transported (defaults to `VelocityStencil`)
-- `divergence_stencil`: same as `vorticity_stencil` but for divergence reconstruction (defaults to `DefaultStencil`)
-- `vertical_scheme`: Scheme used for vertical advection of horizontal momentum. It has to be consistent with the choice of 
-                     `divergence_stencil`. If the latter is a `Nothing`, only `EnergyConservingScheme` is available (this keyword
-                     argument has no effect). In case `divergence_scheme` is an `AbstractUpwindBiasedAdvectionScheme`, 
-                     `vertical_scheme` describes a flux form reconstruction of vertical momentum advection, and any 
-                     advection scheme can be used - `Centered`, `UpwindBiased` and `WENO` (defaults to `EnergyConservingScheme`)
+- `vertical_scheme`: Scheme used for vertical advection of horizontal momentum and upwinding of divergence and kinetic energy gradient. Defaults to `EnergyConservingScheme`)
+- `upwinding`: Treatment of upwinding in case of Upwinding reconstruction of divergence and kinetic energy gradient. Choices are between
+                         `CrossAndSelfUpwinding`, `OnlySelfUpwinding` and `VelocityUpwinding` (defaults to `OnlySelfUpwinding`)
+- `multi_dimensional_stencil` : if true, use a horizontal two dimensional stencil for the reconstruction of vorticity, divergence and kinetic energy gradient.
+                                The tangential direction is _always_ treated with a 5th order centered WENO reconstruction
```

Comment:
```suggestion
- `upwinding`: Treatment of upwinding in case of Upwinding reconstruction of divergence and kinetic energy gradient. Choices are between
                         `CrossAndSelfUpwinding()`, `OnlySelfUpwinding()`, and `VelocityUpwinding()` (defaults to `OnlySelfUpwinding()`).
- `multi_dimensional_stencil` : if `true`, use a horizontal two dimensional stencil for the reconstruction of vorticity, divergence and kinetic energy gradient.
                                The tangential direction is _always_ treated with a 5th-order centered WENO reconstruction.
```